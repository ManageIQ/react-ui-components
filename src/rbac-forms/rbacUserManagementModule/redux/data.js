export const data = {
  rows: [
    {
      id: 100,
      name: 'Adam the first',
      userid: 'Admin',
      email: 'email@neco.com',
      current_group: {
        label: 'Some group',
        onClick: () => console.log('Current group clicked'),
      },
      role: {
        label: 'Admin role',
        onClick: () => console.log('Current role click'),
      },
      groups: [{
        label: 'Cloud-Operators',
        icon: 'group',
        groupId: 1,
        onClick: () => console.log('Group click'),
      }, {
        label: 'Cloud-Users',
        icon: 'group',
        groupId: 2,
        onClick: () => console.log('Group click'),
      }],
      lastlogon: '06/25/18 06:58:14 UTC',
      lastlogoff: '06/30/18 06:58:14 UTC',
    },
    {
      id: 101,
      name: 'Marge the middle',
      userid: 'La transparenza della qualita',
      email: 'email@jandak.com',
      current_group: {
        label: 'Mediocre group',
        onClick: () => console.log('Current group clicked'),
      },
      role: {
        label: 'Shot role',
        onClick: () => console.log('Current role click'),
      },
      groups: [{
        label: 'Cloud-Operators',
        icon: 'group',
        groupId: 1,
        onClick: () => console.log('Group click'),
      }, {
        label: 'Cloud-Users',
        icon: 'group',
        groupId: 2,
        onClick: () => console.log('Group click'),
      }],
      lastlogon: '01/25/18 06:58:14 UTC',
      lastlogoff: '06/30/18 06:58:14 UTC',
    },
    {
      id: 102,
      name: 'Zlatko the last',
      userid: 'Peligro',
      email: 'email@jandak.com',
      current_group: {
        label: 'Admin group',
        onClick: () => console.log('Current group clicked'),
      },
      role: {
        label: 'Cloud-Users',
        onClick: () => console.log('Current role click'),
      },
      groups: [{
        label: 'Cloud-Operators',
        icon: 'group',
        groupId: 1,
        onClick: () => console.log('Group click'),
      }, {
        label: 'Cloud-Users',
        icon: 'group',
        groupId: 2,
        onClick: () => console.log('Group click'),
      }],
      lastlogon: '08/25/18 06:58:14 UTC',
      lastlogoff: '010/30/18 06:58:14 UTC',
    },
  ],
  columns: [{
    property: 'name',
    label: 'Full Name',
  }, {
    property: 'userid',
    label: 'Username',
  }, {
    property: 'email',
    label: 'E-mail',
  }, {
    property: 'current_group',
    label: 'Current Group',
  }, {
    property: 'role',
    label: 'Role',
  }, {
    property: 'lastlogon',
    label: 'Last Logon',
  }, {
    property: 'lastlogoff',
    label: 'Last Logoff',
  }],
  selectedUsers: null,
  groups: [
    {
      value: 1,
      label: 'Hope',
    },
    {
      value: 2,
      label: 'Christa',
    },
    {
      value: 3,
      label: 'Tania',
    },
    {
      value: 4,
      label: 'Mcintyre',
    },
    {
      value: 5,
      label: 'Tonya',
    },
    {
      value: 6,
      label: 'Marcie',
    },
    {
      value: 7,
      label: 'Cara',
    },
    {
      value: 8,
      label: 'Becker',
    },
    {
      value: 9,
      label: 'Foley',
    },
    {
      value: 10,
      label: 'Perkins',
    },
  ],
};
