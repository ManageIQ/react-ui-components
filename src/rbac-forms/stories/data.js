import { action } from '@storybook/addon-actions';

export const groups = [
  {
    value: 1,
    label: 'Hope',
  },
  {
    value: 2,
    label: 'Christa',
  },
  {
    value: 3,
    label: 'Tania',
  },
  {
    value: 4,
    label: 'Mcintyre',
  },
  {
    value: 5,
    label: 'Tonya',
  },
  {
    value: 6,
    label: 'Marcie',
  },
  {
    value: 7,
    label: 'Cara',
  },
  {
    value: 8,
    label: 'Becker',
  },
  {
    value: 9,
    label: 'Foley',
  },
  {
    value: 10,
    label: 'Perkins',
  },
];

export const editedUser = {
  name: 'Users full name',
  userid: 'admin jiny',
  chosen_group: [5, 7],
};

export const usersTableColumns = [{
  property: 'fullname',
  label: 'Full Name',
}, {
  property: 'username',
  label: 'Username',
}, {
  property: 'email',
  label: 'E-mail',
}, {
  property: 'currentgroup',
  label: 'Current Group',
}, {
  property: 'role',
  label: 'Role',
}, {
  property: 'lastlogon',
  label: 'Last Logon',
}, {
  property: 'lastlogoff',
  label: 'Last Logoff',
}];

export const usersTableRows = [
  {
    id: 100,
    fullname: 'Adam the first',
    username: 'Admin',
    email: 'email@neco.com',
    currentgroup: 'Some group',
    role: 'Admin role',
    lastlogon: '06/25/18 06:58:14 UTC',
    lastlogoff: '06/30/18 06:58:14 UTC',
  },
  {
    id: 101,
    fullname: 'Marge the middle',
    username: 'La transparenza della qualita',
    email: 'email@jandak.com',
    currentgroup: 'Mediocre group',
    role: 'Shot role',
    lastlogon: '01/25/18 06:58:14 UTC',
    lastlogoff: '06/30/18 06:58:14 UTC',
  },
  {
    id: 102,
    fullname: 'Zlatko the last',
    username: 'Peligro',
    email: 'email@jandak.com',
    currentgroup: 'Super group',
    role: 'Shot role',
    lastlogon: '08/25/18 06:58:14 UTC',
    lastlogoff: '010/30/18 06:58:14 UTC',
  },
];

export const user = {
  name: 'Administrator',
  userid: 'Admin',
  email: 'mail@mail.com',
  current_group: {
    label: 'EvmGroup-super_administrator',
    onClick: () => action('Current group clicked'),
  },
  groups: [{
    label: 'Cloud-Operators',
    icon: 'group',
    groupId: 1,
    onClick: () => action('Group click'),
  }, {
    label: 'Cloud-Users',
    icon: 'group',
    groupId: 2,
    onClick: () => action('Group click'),
  }],
  role: {
    label: 'Cloud-Users',
    onClick: () => action('Current role click'),
  },
};
