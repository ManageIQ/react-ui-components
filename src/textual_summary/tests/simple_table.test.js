import React from 'react';
import toJson from 'enzyme-to-json';
import { shallow } from 'enzyme';
import SimpleTable from '../simple_table';
import { simpleTableData } from '../data/simple_table';

describe('Simple Table', () => {
  it('renders just fine...', () => {
    /*
     * label:
     *   a) simple
     *   b) object {sortable: true} TODO
     * rows
     *   value:
     *     a) simple
     *     b) object {expandable: true} TODO
     */
    const table = shallow(<SimpleTable
      title={simpleTableData.title}
      labels={simpleTableData.labels}
      rows={simpleTableData.rows}
      onClick={e => null}
    />);
    expect(toJson(table)).toMatchSnapshot();
  });
});
