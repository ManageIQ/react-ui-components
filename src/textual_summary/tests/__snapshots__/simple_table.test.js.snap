// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Simple Table renders just fine... 1`] = `
<table
  className="table table-bordered table-hover table-striped table-summary-screen"
>
  <thead>
    <tr>
      <th
        align="left"
        colSpan={6}
      >
        Firewall Rules
      </th>
    </tr>
    <tr>
      <td
        key="0"
        style={
          Object {
            "wordBreak": "initial",
          }
        }
      >
        <strong>
          Network Protocol
        </strong>
      </td>
      <td
        key="1"
        style={
          Object {
            "wordBreak": "initial",
          }
        }
      >
        <strong>
          Host Protocol
        </strong>
      </td>
      <td
        key="2"
        style={
          Object {
            "wordBreak": "initial",
          }
        }
      >
        <strong>
          Direction
        </strong>
      </td>
      <td
        key="3"
        style={
          Object {
            "wordBreak": "initial",
          }
        }
      >
        <strong>
          Port
        </strong>
      </td>
      <td
        key="4"
        style={
          Object {
            "wordBreak": "initial",
          }
        }
      >
        <strong>
          End Port
        </strong>
      </td>
      <td
        key="5"
        style={
          Object {
            "wordBreak": "initial",
          }
        }
      >
        <strong>
          Source
        </strong>
      </td>
    </tr>
  </thead>
  <tbody>
    <tr
      key="0"
    >
      <td
        className="no-hover"
        key="0"
      />
      <td
        className="no-hover"
        key="1"
      >
        -1
      </td>
      <td
        className="no-hover"
        key="2"
      >
        outbound
      </td>
      <td
        className="no-hover"
        key="3"
      >
        0
      </td>
      <td
        className="no-hover"
        key="4"
      >
        0
      </td>
      <td
        className="no-hover"
        key="5"
      >
        0.0.0.0/0
      </td>
    </tr>
    <tr
      key="1"
    >
      <td
        className="no-hover"
        key="0"
      />
      <td
        className="no-hover"
        key="1"
      >
        TCP
      </td>
      <td
        className="no-hover"
        key="2"
      >
        inbound
      </td>
      <td
        className="no-hover"
        key="3"
      >
        22
      </td>
      <td
        className="no-hover"
        key="4"
      >
        22
      </td>
      <td
        className="no-hover"
        key="5"
      >
        0.0.0.0/0
      </td>
    </tr>
    <tr
      key="2"
    >
      <td
        className="no-hover"
        key="0"
      />
      <td
        className="no-hover"
        key="1"
      >
        TCP
      </td>
      <td
        className="no-hover"
        key="2"
      >
        inbound
      </td>
      <td
        className="no-hover"
        key="3"
      >
        80
      </td>
      <td
        className="no-hover"
        key="4"
      >
        80
      </td>
      <td
        className="no-hover"
        key="5"
      >
        0.0.0.0/0
      </td>
    </tr>
  </tbody>
</table>
`;
